### ISSUE TITLE
`catch-error-name` rule should allow for number postfix

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#93]
Sometimes I need to use a different name so not to shadow an earlier error. This rule should accept a number postfix from `2` to `9`.

For example:

```js
const handleError = err => {
	try {
		options.logger(err);
	} catch (err2) {
		err = new Error('option.logger threw error:\n' + err2.stack);
	}
};
```

---

`_` should be allowed as a name too.

**Comment by SamVerschueren** [comment]
Should the number postfix only be allowed when the variable is shadowed? And if that is the case, should we make sure the number starts with 2 and goes up by 1? Like for example

```js
const handleError = err => {
	try {
		options.logger(err);
	} catch (err9) {
		err = new Error('option.logger threw error:\n' + err9.stack);
	}
};
```

This wouldn't be allowed and should be `err2`.
